### PR TITLE
doc(app-security): how to initilize the security.AppSecurity module (AEROGEAR-8681)

### PR DESCRIPTION
**Pull request by austincunningham**
## Motivation
https://issues.jboss.org/browse/AEROGEAR-8681

## What
Docs for app-security services 

## Why
Requirement for the project

## How
Adding to the Security Service section

## Verification Steps
 
- clone the project 
- checkout my branch AEROGEAR-8681
- change directory into the root of the project 
- need node 8 
- install antora
```bash
npm i -g @antora/cli@1.0
npm i -g @antora/site-generator-default@1.0
mkdir build
```
- run the following script 
```bash
./bin/quick-build.sh local-site.yml
```
- the docs will build in the in the build directory , and are located under the secuity/app security section

docs

### CODE REVIEW

File: modules/ROOT/pages/_partials/app-security/configuring-dev-env-app-security.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
+= Configuring your Development Environment for the {app-security-service} Service
+
+:service-name: App Security
+
+== Downloading the Mobile Services Configuration File
```
Comment (laurafitzgerald) [inline modules/ROOT/pages/_partials/app-security/configuring-dev-env-app-security.adoc]: @austincunningham I think we can import [this file](https://github.com/aerogear/mobile-docs/blob/master/modules/ROOT/pages/_partials/generic-obtaining-the-mobile-sdk-config-file.adoc) here.

Comment (laurafitzgerald) [reply modules/ROOT/pages/_partials/app-security/configuring-dev-env-app-security.adoc]: @finp is there any generic one of these files as we don't need all the platforms here?

Comment (austincunningham) [reply modules/ROOT/pages/_partials/app-security/configuring-dev-env-app-security.adoc]: as discussed that contains more that just cordova so I didn't use it .

**Comment by austincunningham** [comment]
@laurafitzgerald that testing would have been done for the sdk pr https://github.com/aerogear/aerogear-js-sdk/pull/314

File: modules/ROOT/pages/app-security.adoc [hunk 2]
```diff
@@ -27,6 +27,8 @@ ifdef::context[:parent-context: {context}]
 include::_partials/app-security/provisioning-security.adoc[leveloffset=+2]
 include::_partials/app-security/binding-an-app-to-a-service.adoc[leveloffset=+2]
 include::_partials/app-security/configuring-the-service.adoc[leveloffset=+2,tags=!excludeUpstream]
+include::_partials/app-security/setting-up-the-app-security-sdk.adoc[leveloffset=+2]
+include::_partials/app-security/proc-calling-the-app-security-clientInit.adoc[leveloffset=+2]
```
Comment (damienomurchu) [inline modules/ROOT/pages/app-security.adoc]:
```suggestion
== Using the {app-security-service} Service
include::_partials/app-security/proc-calling-the-app-security-clientInit.adoc[leveloffset=+2]
```

Does it make more sense to put this under its own header? ^

File: modules/ROOT/pages/_partials/app-security/configuring-dev-env-app-security.adoc (hunk 1 above)
Comment (finp) [reply modules/ROOT/pages/_partials/app-security/configuring-dev-env-app-security.adoc]: I'm ok with the duplication, if that's the concern here?

File: modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+// end::excludeDownstream[]
+[source,javascript]
+----
+appSecurity.clientInit()
+.then(clientInit => {
+  /**  
+    You can use clientInit.data.disabled boolean 
+    to check if app has been disabled by admin.
+
+    The disabled message will be return if the app is disabled
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc]:
```suggestion
    The disabled message is displayed if the app is disabled 
```

File: modules/ROOT/pages/app-security.adoc (hunk 2 above)
Comment (finp) [reply modules/ROOT/pages/app-security.adoc]: not sure I understand, isn't that heading already in the partial?

File: modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc (hunk 3 above)
Comment (austincunningham) [reply modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc]: Won't be displayed unless the developer/user implements it, I think returned is correct in this case

File: modules/ROOT/pages/app-security.adoc (hunk 2 above)
Comment (austincunningham) [reply modules/ROOT/pages/app-security.adoc]: agree it would be a duplicate heading , maybe set the [leveloffset=+1] which would make it more prominent

Comment (austincunningham) [reply modules/ROOT/pages/app-security.adoc]: had a look and it doesn't look right 
image

will leave it as is

File: modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc (hunk 3 above)
Comment (damienomurchu) [inline modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc]:
```suggestion
    The disabled message will be returned if the app is disabled 
```

File: modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
+= Using the App Security Service
```
Comment (damienomurchu) [inline modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc]:
```suggestion
= Using the {app-security-service} Service
```

File: modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc (hunk 3 above)
Comment (finp) [reply modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc]: gotcha, but also change to present tense, "the message is returned".. makes things easier in the long run

File: modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc (hunk 4 above)
Comment (austincunningham) [reply modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc]: :+1:

File: modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc (hunk 3 above)
Comment (austincunningham) [reply modules/ROOT/pages/_partials/app-security/proc-calling-the-app-security-clientInit.adoc]: :+1:

**Comment by austincunningham** [comment]
@damienomurchu anything needed on this to move forward ?

File: modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+// end::excludeDownstream[]
+Cordova::
++
+--
+
+. Install the link:https://www.npmjs.com/package/@aerogear/security[AeroGear Security] package from link:https://www.npmjs.com/[NPM^]
++
+[source,bash]
+----
+$ npm install @aerogear/security
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc]: Should this include the `--save` argument?

```sh
$ npm install @aerogear/security --save
```

File: modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+[tabs]
+====
+
+// end::excludeDownstream[]
+Cordova::
++
+--
+
+. Install the link:https://www.npmjs.com/package/@aerogear/security[AeroGear Security] package from link:https://www.npmjs.com/[NPM^]
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc]: This should end with a colon to be consistent with the other steps

File: modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+[source,bash]
+----
+$ cordova plugin add @aerogear/cordova-plugin-aerogear-security
+----
+
+. Import and instantiate `SecurityService` to start using the SDK:
++
+[source,javascript]
+----
+const SecurityService = require("@aerogear/security");
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc]: This is an older way of importing I think, and in the showcase app you import like this:

```ts
import { AppSecurity } from '@aerogear/security';
```

Comment (austincunningham) [reply modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc]: Matching the existing doc's

File: modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc (hunk 5 above)
Comment (austincunningham) [reply modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc]: :+1:

File: modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc (hunk 6 above)
Comment (austincunningham) [reply modules/ROOT/pages/_partials/app-security/setting-up-the-app-security-sdk.adoc]: :+1:

**Pull request merged**